### PR TITLE
fix guidance rescale.

### PR DESCRIPTION
**Pull request by entrpn**
Fixes guidance rescale in sdxl where it was erroring out due to traced boolean value.

### CODE REVIEW

File: src/maxdiffusion/generate.py [hunk 1]
```diff
@@ -71,7 +71,12 @@ def loop_body(step, args, model, pipeline, prompt_embeds, guidance_scale, guidan
     noise_pred = noise_pred_uncond + guidance_scale * (noise_prediction_text - noise_pred_uncond)
 
     # Based on 3.4. in https://arxiv.org/pdf/2305.08891.pdf
-    noise_pred = rescale_noise_cfg(noise_pred, noise_prediction_text, guidance_rescale=guidance_rescale)
```
Comment (parambole) [inline src/maxdiffusion/generate.py:74]: nit: 
Adding a comment around what is being done "This technique aims to improve image quality by adjusting the noise prediction."

File: src/maxdiffusion/generate_sdxl.py [hunk 2]
```diff
@@ -68,15 +68,20 @@ def loop_body(step, args, model, pipeline, added_cond_kwargs, prompt_embeds, gui
...
   if config.do_classifier_free_guidance:
-    noise_pred = apply_classifier_free_guidance(noise_pred, guidance_scale, guidance_rescale)
+    noise_pred, noise_prediction_text = apply_classifier_free_guidance(noise_pred, guidance_scale)
+
+  noise_pred = jax.lax.cond(
+      guidance_rescale[0] > 0,
+      lambda _: rescale_noise_cfg(noise_pred, noise_prediction_text, guidance_rescale),
+      lambda _: noise_pred,
+      operand=None
+  )
```
Comment (parambole) [inline src/maxdiffusion/generate_sdxl.py:87]: nit: Since this call is being used in both of the files is there a way to put it in a common utils file ?

Comment (entrpn) [reply src/maxdiffusion/generate_sdxl.py:87]: I don't think the jax conditional needs to be refactored. The main code is in the rescale_noise_cfg fn.

File: src/maxdiffusion/generate.py (hunk 1 above)
Comment (entrpn) [reply src/maxdiffusion/generate.py:74]: Added comments to explain what this does.

**Review by parambole — APPROVED** [review]

**Pull request merged**